autoload/kite/languages.vim

```diff
@@ -0,0 +1,32 @@
+let s:languages_supported_by_kited = []
+
+" Returns true if the current buffer's language is supported by this plugin, false otherwise.
+function! kite#languages#supported_by_plugin()
+  if &filetype == 'python' && expand('%:e') != 'pyi' && index(g:kite_supported_languages, 'python') != -1
+    return 1
+  endif
+
+  if &filetype == 'go' && index(g:kite_supported_languages, 'go') != -1
+    return 1
+  endif
+
+  return 0
+endfunction
+
+
+" Returns true if the current buffer's language is supported by kited, false otherwise.
+function! kite#languages#supported_by_kited()
+  " Only check kited's languages once.
+  if empty(s:languages_supported_by_kited)
+    let s:languages_supported_by_kited = kite#client#languages(function('kite#languages#handler'))
```

Comment:
Which set of languages is assumed when this requests fails?

Reply:
No languages, but the code will keep trying kited (when entering a buffer) until it receives a non-empty response.